op-service/event/system.go

```diff
@@ -80,6 +92,34 @@ type systemActor struct {
 	emitPriority Priority
 }
 
+func (r *systemActor) traceAndLogEventEmitted(ctx context.Context, ev Event) context.Context {
+	_, path, line, _ := runtime.Caller(2)
+	if strings.Contains(path, "limiter.go") {
+		_, path, line, _ = runtime.Caller(3)
+	}
+
+	file := filepath.Base(path)
+	dir := filepath.Base(filepath.Dir(path))
+
+	var euuid string
+	var estep int
+	if ctx.Value(ctxKeyEventUuid) == nil {
+		euuid = uuid.New().String()[:6]
+		estep = 0
+		ctx = context.WithValue(ctx, ctxKeyEventUuid, euuid)
+		ctx = context.WithValue(ctx, ctxKeyEventStep, estep)
+	} else {
+		euuid = ctx.Value(ctxKeyEventUuid).(string)
```

Comment:
(nit) We may use 
```
type eventTrace struct {
    UUID string
    Step int
}
```
and attach the struct to the context. We may extend by adding extra fields in the future.

Also, can we sanity check like
```
euuid, ok = ctx.Value(ctxKeyEventUuid).(string)
```
and check the ok just in case to avoid the panic?